platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -1165,4 +1175,340 @@ private interface AddressBookCallback {
+                                "Certificate for node {} does not match the migrated certificate",
+                                nodeId);
+                        errorCount.incrementAndGet();
+                    }
+                } catch (CertificateEncodingException e) {
+                    logger.error(ERROR.getMarker(), "Encoding error while validating certificate for node {}.", nodeId);
+                    errorCount.incrementAndGet();
+                }
+            }
+        });
+        return errorCount.get();
+    }
+
+    /**
+     * Rollback the creation of PEM files for signing keys and certificates.
+     *
+     * @param pfxPrivateKeys  the map of extracted private keys.
+     * @param pfxCertificates the map of extracted certificates.
+     * @throws KeyStoreException   if the underlying method calls throw this exception.
+     * @throws KeyLoadingException if the underlying method calls throw this exception.
+     */
+    private void rollBackSigningKeysAndCertsChanges(
+            final Map<NodeId, PrivateKey> pfxPrivateKeys, final Map<NodeId, Certificate> pfxCertificates)
+            throws KeyStoreException, KeyLoadingException {
+
+        AtomicLong cleanupErrorCount = new AtomicLong(0);
+        iterateAddressBook(addressBook, (i, nodeId, address, nodeAlias) -> {
+            // private key rollback
+            if (isLocal(address) && pfxPrivateKeys.containsKey(address.getNodeId())) {
+                try {
+                    Files.deleteIfExists(privateKeyStore(nodeAlias, KeyCertPurpose.SIGNING));
+                } catch (IOException e) {
+                    cleanupErrorCount.incrementAndGet();
+                }
+            }
+            // certificate rollback
+            if (pfxCertificates.containsKey(address.getNodeId())) {
+                try {
+                    Files.deleteIfExists(certificateStore(nodeAlias, KeyCertPurpose.SIGNING));
+                } catch (IOException e) {
+                    cleanupErrorCount.incrementAndGet();
+                }
+            }
+        });
+        if (cleanupErrorCount.get() > 0) {
+            logger.error(
+                    ERROR.getMarker(),
+                    "Failed to rollback {} pem files created. Manual cleanup required.",
+                    cleanupErrorCount.get());
+            throw new IllegalStateException("Cryptography Migration failed to generate or validate PEM files.");
+        }
+    }
+
+    /**
+     * Move the PFX files to the OLD_PFX_KEYS sub-directory.
+     *
+     * @throws KeyStoreException   if the underlying method calls throw this exception.
+     * @throws KeyLoadingException if the underlying method calls throw this exception.
+     */
+    private void cleanupMovePfxFilesToSubDirectory() throws KeyStoreException, KeyLoadingException {
+        AtomicLong cleanupErrorCount = new AtomicLong(0);
+
+        final String now = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH:mm:ss").format(LocalDateTime.now());
```

Comment:
Minor: it may make it difficult to work with files/directories that have `:` in their name because this character needs to be escaped. You may want to consider replacing `:` with, say, `_`, or instead add `h`, `m`, and `s` after the time numbers.

Reply:
updated to remove colons